src/com/carolinarollergirls/scoreboard/defaults/DefaultMediaModel.java

```diff
@@ -0,0 +1,235 @@
+            }
+        }
+    }
+
+    public Set<String> getFormats() {
+        synchronized (coreLock) {
+            return Collections.unmodifiableSet(files.keySet());
+        }
+    }
+
+    public Set<String> getTypes(String format) {
+        synchronized (coreLock) {
+            return Collections.unmodifiableSet(files.get(format).keySet());
+        }
+    }
+
+    public Map<String, MediaFile> getMediaFiles(String format, String type) {
+        return Collections.unmodifiableMap(new HashMap<String, MediaFile>(getMediaFileModels(format, type)));
+    }
+
+    public Map<String, MediaFileModel> getMediaFileModels(String format, String type) {
+        Map<String, Map<String, MediaFileModel>> fm = files.get(format);
+        if (fm == null) {
+            return null;
+        }
+        Map<String, MediaFileModel> tm = fm.get(type);
+        if (tm == null) {
+            return null;
+        }
+        return Collections.unmodifiableMap(tm);
+    }
+
+    public boolean removeMediaFile(String format, String type, String id) {
+        synchronized (coreLock) {
+            // Check the directory is one the user is meant to be able to change.
+            if (files.containsKey(format) && files.get(format).containsKey(type)) {
+                // Delete the file, and let inotify take care of handling the change.
+                try {
+                    return Files.deleteIfExists(path.resolve(format).resolve(type).resolve(id));
+                } catch (IOException e) {
+                    return false;
+                }
+            }
+            return false;
+        }
+    }
+
+    private Map<String, Map<String, Map<String, MediaFileModel>>> files = new HashMap<String, Map<String, Map<String, MediaFileModel>>>();
+    private Path path;
+    private WatchService watchService;
+
+    private static Object coreLock = DefaultScoreBoardModel.getCoreLock();
+
+    public class DefaultMediaFileModel extends DefaultScoreBoardEventProvider implements MediaFileModel {
+        DefaultMediaFileModel(String format, String type, String id, String name, String src) {
+            this.format = format;
+            this.type = type;
+            this.id = id;
+            this.name = name;
+            this.src = src;
+        }
+
+        public String getProviderName() { return "MediaFile"; }
+        public Class<MediaFile> getProviderClass() { return MediaFile.class; }
+        public String getProviderId() { return getId(); }
+
+        public String getFormat() { synchronized (coreLock) { return format ;} }
+        public String getType() { synchronized (coreLock) { return type ;} }
+        public String getId() { synchronized (coreLock) { return id ;} }
+        public String getName() { synchronized (coreLock) { return name ;} }
+        public void setName(String n) {
+            synchronized (coreLock) {
+                name = n;
+                scoreBoardChange(new ScoreBoardEvent(this, EVENT_FILE, this, null));
```

Comment:
This should probably have it's own event, so we aren't forced to replace the whole element in order to change one attribute.

Reply:
It's one tr, and not on a critical path. I don't think it's worth it.